Python/pylifecycle.c

```diff
@@ -1835,8 +1836,13 @@ Py_FinalizeEx(void)
      * Alas, a lot of stuff may still be alive now that will be cleaned
      * up later.
      */
+
+    FILE *fp = stderr;
+    if (python_dump_file != NULL) {
+        fp = _Py_wfopen(python_dump_file, L"w");
```

Comment:
Why not calling _Py_PrintReferences()+fclose() here? It would allow to use the 2 env vars separately.

I'm not sure how your current PR is supposed to be used. Do you have to set both env vars? `PYTHONDUMPREFS=1 PYTHONDUMPFILE=dump.txt`? I would prefer to be able only specify `PYTHONDUMPFILE=dump.txt`.

You should check for failure: fp == NULL. Maybe just log an error? For example, if the current user cannot write in the current directory.